DQM/SiPixelPhase1TrackClusters/src/SiPixelPhase1TrackClusters.cc

```diff
@@ -21,25 +21,39 @@
 #include "DataFormats/TrackReco/interface/Track.h"
 #include "DataFormats/TrackerRecHit2D/interface/SiPixelRecHit.h"
 #include "TrackingTools/TrackFitters/interface/TrajectoryStateCombiner.h"
+#include "DataFormats/VertexReco/interface/Vertex.h"
 
 
 SiPixelPhase1TrackClusters::SiPixelPhase1TrackClusters(const edm::ParameterSet& iConfig) :
   SiPixelPhase1Base(iConfig) 
 {
   clustersToken_ = consumes<edmNew::DetSetVector<SiPixelCluster>>(iConfig.getParameter<edm::InputTag>("clusters"));
+
   tracksToken_ = consumes<reco::TrackCollection>(iConfig.getParameter<edm::InputTag>("tracks"));
+
+  offlinePrimaryVerticesToken_ = consumes<reco::VertexCollection>(std::string("offlinePrimaryVertices"));
+
+  applyVertexCut_=iConfig.getUntrackedParameter<bool>("VertexCut",true);
 }
 
 void SiPixelPhase1TrackClusters::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetup) {
 
+  if (! (histo.size() > ONTRACK_SIZE_VS_ETA)) return;
```

Comment:
this is a QUICK fix.
we may have to issue a LogWarning in future  (actually an assert)